INSS.EIIR.Functions/Functions/ExtractJobNotificationTrigger.cs

```diff
@@ -0,0 +1,87 @@
+{
+    public class ExtractJobNotificationTrigger
+    {
+        private readonly ILogger<ExtractJobNotificationTrigger> _logger;
+        private readonly IServiceBusMessageSender _serviceBusMessageSender;
+        private readonly IExtractRepository _eiirRepository;
+        private readonly ServiceBusConfig _serviceBusConfig;
+        private readonly NotifyConfig _notifyConfig;
+
+        public ExtractJobNotificationTrigger(
+            ILogger<ExtractJobNotificationTrigger> log,
+            IOptions<ServiceBusConfig> sbOptions,
+            IOptions<NotifyConfig> nfOptions,
+            IServiceBusMessageSender serviceBusMessageSender,
+            IExtractRepository eiirRepository)
+        {
+            _logger = log;
+            _serviceBusConfig = sbOptions.Value;
+            _notifyConfig = nfOptions.Value;
+            _serviceBusMessageSender = serviceBusMessageSender;
+            _eiirRepository = eiirRepository;
+        }
+
+        [FunctionName("ExtractJobNotificationTrigger")]
+        public async Task Run([BlobTrigger("%blob:containername%/{name}.zip", Connection = "blob:connectionstring")] byte[] myBlob, string name, Uri uri)
+        {
+            string message = $"ExtractJobNotificationTrigger Blob trigger function triggered for blob\n Name: {name}  \n with uri:{uri.AbsoluteUri}";
+            _logger.LogInformation(message);
+
+            if (string.IsNullOrEmpty(_serviceBusConfig.NotifyQueue))
+            {
+                _logger.LogError("ExtractjobTrigger missing configuration servicebus:notifyqueue");
+                throw new Exception("ExtractjobTrigger missing configuration servicebus:notifyqueue");
+            }
+
+            if (string.IsNullOrEmpty(_notifyConfig.SubscriberEmailTemplateId))
+            {
+                _logger.LogError("ExtractjobTrigger missing configuration subscriberEmailTemplateId");
+                throw new Exception("ExtractjobTrigger missing configuration subscriberEmailTemplateId");
+            }
+
+            // 1. Get a list of Active subscribers
+            var activeSubscribers = await _eiirRepository.GetActiveSubscribers();
+
+            // 2. Send Notification message that subscriber file has been generated
+            foreach (var subscriber in activeSubscribers)
+            {
+                var subscriberDetails = await _eiirRepository.GetSubscriberDetails(subscriber.SubscriberId);
+                if (subscriberDetails == null)
+                {
+                    _logger.LogError($"Subscribed details not found for subscriber Id: [{subscriber.SubscriberId}]");
+                    throw new Exception($"Subscribed details not found for subscriber Id: [{subscriber.SubscriberId}]");
+                }
+
+                var body = new
+                {
+                    TemplateId = _notifyConfig.SubscriberEmailTemplateId,
+                    ApiKey = _notifyConfig.ApiKey,
+                    EmailAddress = subscriberDetails.ContactEmail,
```

Comment:
old school ... hate dynamic unless absolutely necessary ... feels like it could be string

Reply:
again this is down to gov notify personalisation object which has it defined as dynamic, I assume for flexibility, here's the documentation from notify. So for placeholders in email templates we have to pass the following:

personalisation (optional)
If a template has placeholder fields for personalised information such as name or reference number, you need to provide their values in a Dictionary. For example:

Dictionary<String, dynamic> personalisation: new Dictionary<String, dynamic>
{
    { "first_name", "Amala"
      "application_date", "2018-01-01"
    }
};